LivrETS/Scripts/managefairs.js

```diff
@@ -16,19 +16,68 @@ You should have received a copy of the GNU General Public License
 along with this program.  If not, see <http://www.gnu.org/licenses/>.
 */
 $(document).ready(function () {
-    $("thead>tr").find("th").addClass("text-center");
+    //init managefairs table width datables plugins
+
+    $('table').DataTable({
+        processing: true,
+        ajax: {
+            url: "/Admin/ListFairs",
+            type: "POST",
+            dataType: "JSON",
+            dataSrc: function (val) {
+
+                return val.listFairs
+            }
+        },
+        columns: [
+            {
+                class: "check-row text-center",
+                sortable: false,
+                data: function (val) {
+                    return "<input type='checkbox' name='check-select-fair' data-fair-id='" + val.Id + "' />";
+                }
+            },
+            {
+                data: "LivrETSID",
+                visible: false
+            },
+            {
+                data: function (val) {
+                    return new Date(parseInt(val.StartDate.replace('/Date(', ''))).toDateString();
+                }
+            },
+            {
+                data: function (val) {
+                    return new Date(parseInt(val.EndDate.replace('/Date(', ''))).toDateString();
+                }
+            },
+            {
+                data: "Phase"
+            },
+            {
+                class: "text-center",
+                sortable: false,
+                data: function (val) {
+                    return "<a href='#' class='btn btn-sm btn-primary btn-edit-fair' data-fair-id='"+ val.Id +"' ><span class='glyphicon glyphicon-edit'></span></a> "+
+                    "<a href='#' class='btn btn-sm btn-danger btn-delete-fair' data-fair-id='" + val.Id + "'><span class='glyphicon glyphicon-trash'></span></a>"
+                    
+                }
+            }
+        ]
+
+    });
 
     $("#close-error").on("click", function () {
         $("#errors").hide("slow");
     });
 
     // Select single fair event.
-    $("input[type='checkbox'][name='check-select-fair']").on("change", function () {
+    $('table tbody').on("click", ".btn-edit-fair", function () {
         updateDeleteSelectedView();
```

Comment:
Ok

Le 12 août 2016 21:56, "Charles Levesque" <notifications@github.com> a
écrit :

> In LivrETS/Scripts/managefairs.js
> <https://github.com/Tributerre/LivrETS-Web/pull/38#discussion_r74679519>:
>
> >
> >      $("#close-error").on("click", function () {
> >          $("#errors").hide("slow");
> >      });
> >
> >      // Select single fair event.
> > -    $("input[type='checkbox'][name='check-select-fair']").on("change", function () {
> > +    $('table tbody').on("click", ".btn-edit-fair", function () {
> >          updateDeleteSelectedView();
>
> C'est tout à fait attendu. Il y a une proposition de changement sur la
> disposition des boutons d'actions qui sont dans un panneau sur le côté pour
> les disposer dans le haut du tableau de données comme dans Gmail. Par
> contre, c'est pour une version ultérieure.
>
> —
> You are receiving this because you authored the thread.
> Reply to this email directly, view it on GitHub
> <https://github.com/Tributerre/LivrETS-Web/pull/38/files/826f90f968aab172d7dcabf4bcc7175fd3dc049f#r74679519>,
> or mute the thread
> <https://github.com/notifications/unsubscribe-auth/AFC5jVqw8PIcmnxsUfOedve0x7nQx6wlks5qfSRmgaJpZM4JZMmE>
> .
>